ext/SpeedyWeatherFiniteDifferencesExt.jl

```diff
@@ -27,4 +27,79 @@ function FiniteDifferences.to_vec(x::LTA) where LTA <: LowerTriangularArray
+
+    v, vals_from_vec = to_vec(vals)
+    function structtype_from_vec(v::Vector{<:Real})
+        val_vecs = vals_from_vec(v)
+        values = map((b, v) -> b(v), backs, val_vecs)
+        
+        T(vals_excluded_pre..., values..., vals_excluded_post...)
+    end
+    return v, structtype_from_vec
+end
+
+included_fields(::Type{<:PrognosticVariables}) = (:vor, :div, :temp, :humid, :pres, :random_pattern, :ocean, :land, :tracers, :particles)
+excluded_fields_pre(::Type{<:PrognosticVariables}) = (:trunc, :nlat_half, :nlayers, :nparticles)
+excluded_fields_post(::Type{<:PrognosticVariables}) = (:scale, :clock)
+
+included_fields(::Type{<:PrognosticVariablesOcean}) = (:sea_surface_temperature, :sea_ice_concentration)
+excluded_fields_pre(::Type{<:PrognosticVariablesOcean}) = (:nlat_half, )
+excluded_fields_post(::Type{<:PrognosticVariablesOcean}) = ()
+
+included_fields(::Type{<:PrognosticVariablesLand}) = (:land_surface_temperature, :snow_depth, :soil_moisture_layer1, :soil_moisture_layer2)
+excluded_fields_pre(::Type{<:PrognosticVariablesLand}) = (:nlat_half, )
+excluded_fields_post(::Type{<:PrognosticVariablesLand}) = ()
+
+included_fields(::Type{<:DiagnosticVariables}) = (:tendencies, :grid, :dynamics, :physics, :particles, :column, :temp_average)
+excluded_fields_pre(::Type{<:DiagnosticVariables}) = (:trunc, :nlat_half, :nlayers, :nparticles)
+excluded_fields_post(::Type{<:DiagnosticVariables}) = (:scale,)
+
+included_fields(::Type{<:Tendencies}) = (:vor_tend, :div_tend, :temp_tend, :humid_tend, :u_tend, :v_tend, :pres_tend, :tracers_tend, :u_tend_grid, :v_tend_grid, :temp_tend_grid, :humid_tend_grid, :pres_tend_grid, :tracers_tend_grid)
+excluded_fields_pre(::Type{<:Tendencies}) = (:trunc, :nlat_half, :nlayers)
+excluded_fields_post(::Type{<:Tendencies}) = ()
+
+included_fields(::Type{<:GridVariables}) = (:vor_grid, :div_grid, :temp_grid, :temp_virt_grid, :humid_grid, :u_grid, :v_grid, :pres_grid, :tracers_grid, :random_pattern, :temp_grid_prev, :humid_grid_prev, :u_grid_prev, :v_grid_prev, :pres_grid_prev, :tracers_grid_prev)
+excluded_fields_pre(::Type{<:GridVariables}) = (:nlat_half, :nlayers)
+excluded_fields_post(::Type{<:GridVariables}) = ()
+
+included_fields(::Type{<:DynamicsVariables}) = (:a, :b, :a_grid, :b_grid, :a_2D, :b_2D, :a_2D_grid, :b_2D_grid, :uv∇lnp, :uv∇lnp_sum_above, :div_sum_above, :temp_virt, :geopot, :σ_tend, :∇lnp_x, :∇lnp_y, :u_mean_grid, :v_mean_grid, :div_mean_grid, :div_mean)
+excluded_fields_pre(::Type{<:DynamicsVariables}) = (:trunc, :nlat_half, :nlayers)
+excluded_fields_post(::Type{<:DynamicsVariables}) = ()
+
+included_fields(::Type{<:PhysicsVariables}) = (:precip_large_scale, :precip_convection, :precip_rate_large_scale, :precip_rate_convection, :cloud_top, :soil_moisture_availability, :sensible_heat_flux, :evaporative_flux, :surface_shortwave_up, :surface_shortwave_down, :surface_longwave_up, :surface_longwave_down, :outgoing_shortwave_radiation, :outgoing_longwave_radiation, :cos_zenith)
```

Comment:
these all have to be hardcoded? Because I'm frequently changing these fields. But note that everything in `diagnostic_variables.physics` is actually only stored from the parameterizations for output. None of these values feed back into the model. Well here in this version they do but with #667 merged I moved the prescribed heat fluxes into the prognostic variables as they do influence the simulation.

Reply:
Yeah, I totally see that this is maintenance heavy code to do it like this. It was a quick thing to code it like that. 

Let's see how could we do this different. I think this could probably also be just based on the types of the fields. E.g. exclude `Int`, include all `AbstractArray` and container formats.

Reply:
I committed a much less hardcoded version. The types are still hardcoded though.